elip-0xxx.mediawiki

```diff
@@ -0,0 +1,150 @@
+
+The optional mempool policies proposed here give sidechain operators configurable options to enable a peg-in subsidy from a specific chain height, an amount threshold above which no subsidy is required, and a minimum peg-in amount from a specified height.
+
+These options allow the operator to ensure that new outputs from peg-ins subsidize the cost of their own consolidation, and encourage users to use alternative options like atomic swaps for amounts below the subsidy threshold.
+
+==Design==
+
+===Overview===
+
+New optional configurable chain parameters are added to Elements: peg-in subsidy height, peg-in subsidy threshold amount, peg-in minimum height, and peg-in minimum amount. Their default values leave them disabled unless specified. If a subsidy height is set, then from that height a peg-in transaction must include a subsidy output if the value of the peg-in is below the subsidy threshold value. Independently, if a minimum peg-in amount is set then any peg-in value must be greater than that minimum amount, after the peg-in minimum block height.
+
+A subsidy output is an <code>OP_RETURN</code> output in the sidechain claim transaction that burns a value equal to spending the new parent output at the parent transaction fee rate. Elements RPCs <code>claimpegin</code> and <code>createrawpegin</code> are modified to create these subsidy outputs when required, and mempool policy validation is added to check for subsidies and minimum peg-in amounts as configured.
+
+===Specification===
+
+====Node====
+
+New options are added to Elements configuration: <code>peginsubsidyheight</code>, <code>peginsubsidythreshold</code>, <code>peginminheight</code>, and <code>peginminamount</code>. By default, these are unset. For <code>peginsubsidythreshold</code> and <code>peginminamount</code> this is an amount of 0, and for <code>peginsubsidyheight</code> and <code>peginminheight</code> this is the value of int32 max.
+
+New mempool validation is implemented to check for a peg-in subsidy from the subsidy height. This requires that a peg-in transaction, whose value is below the subsidy threshold, must contain one or more <code>OP_RETURN</code> outputs burning the value of spending a P2WSH output on the parent chain at the same fee rate as the parent peg-in. A new policy will also reject peg-ins below the minimum peg-in amount after the minimum peg-in height.
+
+This validation requires that the sidechain node is validating peg-ins against the parent node, as the sidechain node calls <code>getrawtransaction</code> to retrieve the fee paid and vsize of the parent transaction to calculate its fee rate. If the sidechain node is not validating peg-ins (<code>validatepegin=0</code>), then it will accept the transaction without subsidy check, but this transaction will fail to be relayed to nodes that are validating peg-ins. The minimum acceptable parent fee rate is 1 sat/vb.
```

Comment:
In 10a67c39a41e08dd11e6ca346a28313db108a8e1:

Why have this rule? If a pegin was accepted on the Bitcoin chain at a lower feerate then why are we second-guessing that? Also, if we want to enforce a minimum, why set it by forbidding certain parent transactions (which I guess would then just burn coins, or...?) instead of setting an alternate minimum on the size of the OP_RETURN output?